src/wallet/rpcwallet.cpp

```diff
@@ -2347,6 +2347,74 @@ static RPCHelpMan settxfee()
+        RPCExamples{
+            ""
+            "\nSet a fee rate of 1 " + CURRENCY_ATOM + "/vB\n"
+            + HelpExampleCli("setfeerate", "1") +
+            "\nSet a fee rate of 3.141 " + CURRENCY_ATOM + "/vB\n"
+            + HelpExampleCli("setfeerate", "3.141") +
+            "\nSet a fee rate of 7.75 " + CURRENCY_ATOM + "/vB with named arguments\n"
+            + HelpExampleCli("-named setfeerate", "amount=7.75") +
+            "\nSet a fee rate of 25 " + CURRENCY_ATOM + "/vB with the RPC\n"
+            + HelpExampleRpc("setfeerate", "25")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            std::shared_ptr<CWallet> const rpc_wallet{GetWalletForJSONRPCRequest(request)};
+            if (!rpc_wallet) return NullUniValue;
+            CWallet& wallet = *rpc_wallet;
+
+            LOCK(wallet.cs_wallet);
+
+            CFeeRate amount{AmountFromValue(request.params[0]), COIN};
+            std::string amount_str{amount.ToString(FeeEstimateMode::SAT_B)};
+            std::string current_amount_str{wallet.m_pay_tx_fee.ToString(FeeEstimateMode::SAT_B)};
+            CFeeRate relay_min_feerate(wallet.chain().relayMinFee().GetFeePerK());
+            CFeeRate wallet_min_feerate(wallet.m_min_fee.GetFeePerK());
+            CFeeRate wallet_max_feerate(wallet.m_default_max_tx_fee, 1000);
+            std::string result, error;
+
+            if (amount == CFeeRate(0)) {
+                wallet.m_pay_tx_fee = amount;
+                result = "Fee rate for transactions with this wallet successfully unset. By default, automatic fee selection will be used.";
+            } else if (amount < relay_min_feerate) {
+                error = strprintf("The requested fee rate of %s cannot be less than the minimum relay fee rate of %s. The current setting of %s for this wallet remains unchanged.", amount_str, relay_min_feerate.ToString(FeeEstimateMode::SAT_B), current_amount_str);
+            } else if (amount < wallet_min_feerate) {
+                error = strprintf("The requested fee rate of %s cannot be less than the wallet min fee rate of %s. The current setting of %s for this wallet remains unchanged.", amount_str, wallet_min_feerate.ToString(FeeEstimateMode::SAT_B), current_amount_str);
+            } else if (amount > wallet_max_feerate) {
+                error = strprintf("The requested fee rate of %s cannot be greater than the wallet max fee rate of %s. The current setting of %s for this wallet remains unchanged.", amount_str, wallet_max_feerate.ToString(FeeEstimateMode::SAT_B), current_amount_str);
+            } else {
+                wallet.m_pay_tx_fee = amount;
+                result = "Fee rate for transactions with this wallet successfully set to " + amount_str;
+            }
+
+            UniValue obj(UniValue::VOBJ);
+            obj.pushKV("wallet_name", wallet.GetName());
+            obj.pushKV("fee_rate", UniValue(UniValue::VNUM, wallet.m_pay_tx_fee.SatsToString()));
```

Comment:
IMO we should add a `ValueFromFeeRate`

Reply:
Maybe rebase this on top of 82b0352144fbe21b60a7d3f9351debec758618a0

Reply:
> Maybe rebase this on top of [8798383](https://github.com/bitcoin/bitcoin/commit/8798383475f4e1db44f734cdd94885e5527761fe)

SGTM

Reply:
> Maybe rebase this on top of [8798383](https://github.com/bitcoin/bitcoin/commit/8798383475f4e1db44f734cdd94885e5527761fe)

Pulled 8798383 in as commit 0e2c3ace86b8aa